stubs/10.0.0/HasManyThrough.stub

```diff
@@ -48,6 +48,34 @@ class HasManyThrough extends Relation
      */
     public function cursorPaginate($perPage = null, $columns = ['*'], $cursorName = 'cursor', $cursor = null);
 
+    /**
+     * Execute the query and get the first related model.
+     *
+     * @param  array<int, string>  $columns
+     * @return TRelatedModel|null
+     */
+    public function first($columns = ['*']);
+
+    /**
+     * Execute the query and get the first result or throw an exception.
+     *
+     * @param  array<int, string>  $columns
+     * @return TRelatedModel
+     *
+     * @throws \Illuminate\Database\Eloquent\ModelNotFoundException<TRelatedModel>
+     */
+    public function firstOrFail($columns = ['*']);
+
+    /**
+     * Execute the query and get the first result or call a callback.
+     *
+     * @template TReturnClosure
+     * @param  \Closure(): TReturnClosure|array<int, string>  $columns
```

Comment:
All of these closure generics are incorrect (same for other stub), they should be

```php
     * @param  (\Closure(): TReturnClosure)|array<int, string>  $columns
     * @param  (\Closure(): TReturnClosure)|null  $callback
```

Reply:
I think that at least PHPStan assumes the braces otherwise the tests would fail, but I agree that your way is more readable. I just pushed the changes.